controllers/nodehealthcheck_controller.go

```diff
@@ -496,8 +542,23 @@ func (r *NodeHealthCheckReconciler) patchStatus(nhc, nhcOrig *remediationv1alpha
 	return r.Client.Status().Patch(context.Background(), nhc, mergeFrom, &client.PatchOptions{})
 }
 
-func (r *NodeHealthCheckReconciler) getInflightRemediations(nhc *remediationv1alpha1.NodeHealthCheck, template *unstructured.Unstructured) (map[string]metav1.Time, error) {
-	cr, err := r.generateRemediationCR(&v1.Node{}, nhc, template)
+func (r *NodeHealthCheckReconciler) getOwnedInflightRemediations(nhc *remediationv1alpha1.NodeHealthCheck, template *unstructured.Unstructured) (map[string]metav1.Time, error) {
+	all, err := r.getAllInflightRemediations(template)
+	if err != nil {
+		return nil, err
+	}
+	owned := make(map[string]metav1.Time)
+	for _, remediationCR := range all {
+		if isOwner(&remediationCR, nhc) {
+			owned[remediationCR.GetName()] = remediationCR.GetCreationTimestamp()
+			continue
+		}
+	}
+	return owned, nil
+}
+
+func (r *NodeHealthCheckReconciler) getAllInflightRemediations(template *unstructured.Unstructured) ([]unstructured.Unstructured, error) {
+	cr, err := r.generateRemediationCR(&v1.Node{}, nil, template)
```

Comment:
I have a couple of questions :-)

1. From  `getAllInflightRemediations` method name I'd expect it to retrieve existing remediations  and not create them, but IIUC it calls  `generateRemediationCR` that does just that.

2. I'm not completely clear on nhc role in `generateRemediationCR` and why/when it's ok for it to be nil.

Reply:
Good questions, I had to look it up while refactoring this as well 😎 

1. We need an example CR for the UnstructeredList, so it know the type of its items. The CR is not created in the cluster, just in memory.
2. For the above goal, the owner ref on the generated CR is uninteresting. We just need the type information.